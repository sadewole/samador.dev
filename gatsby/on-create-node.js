const { createFilePath } = require(`gatsby-source-filesystem`)

const onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions

  if (node.internal.type === `MarkdownRemark`) {
    if (typeof node.frontmatter.slug !== "undefined") {
      const dirname = getNode(node.parent).relativeDirectory
      createNodeField({
        node,
        name: "slug",
        value: `/${dirname}/${node.frontmatter.slug}`,
      })
    } else {
      const value = createFilePath({ node, getNode })
      createNodeField({
        node,
        name: "slug",
        value,
      })
    }
  }
}

module.exports = onCreateNode
