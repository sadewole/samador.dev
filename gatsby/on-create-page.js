const path = require(`path`)
const createBlogs = require("./pagination/create-blogs")

const createPages = async ({ graphql, actions, reporter }) => {
  const { createPage } = actions

  // Posts and pages from markdown
  const result = await graphql(`
    {
      allMarkdownRemark(filter: { frontmatter: { draft: { ne: true } } }) {
        nodes {
          frontmatter {
            slug
            layout
            title
          }
        }
      }
    }
  `)

  if (result.errors) {
    reporter.panicOnBuild(`There was an error loading this page`, result.errors)
    return
  }

  const { nodes } = result.data.allMarkdownRemark

  nodes.forEach(node => {
    if (node.frontmatter.layout === "page") {
      createPage({
        path: node.frontmatter.slug,
        component: path.resolve("./src/templates/page-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    } else if (node.frontmatter.layout === "blog") {
      createPage({
        path: `/blogs/${node.frontmatter.slug}`,
        component: path.resolve("./src/templates/blog-template.js"),
        context: { slug: node.frontmatter.slug },
      })
    }
  })

  await createBlogs(graphql, actions)
}

module.exports = createPages
