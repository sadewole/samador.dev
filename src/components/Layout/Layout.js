import React from "react"
import PropTypes from "prop-types"
import Seo from "../common/Seo"
import "./layout.css"

const Layout = ({ title, children }) => {
  return (
    <div className="py-10 sm:px-10 bg-gray-700 text-gray-300 dark:bg-gray-900 m-0 min-h-screen overflow-x-hidden">
      <Seo title={title} />
      <div className="max-w-6xl xl:w-10/12  m-auto">{children}</div>
    </div>
  )
}

Layout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
}

export default Layout
