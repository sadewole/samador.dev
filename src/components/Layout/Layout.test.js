import React from "react"
import { StaticQuery, useStaticQuery } from "gatsby"
import renderer from "react-test-renderer"

import siteMetaData from "../../../jest/__fixtures__/site-metadata"
import Layout from "./Layout"

describe("Layout", () => {
  const props = {
    ...siteMetaData,
    title: "test",
    children: "test",
  }

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }) => render(props),
      useStaticQuery.mockReturnValue(props)
    )
  })

  test("render correctly", () => {
    const tree = renderer.create(<Layout {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
