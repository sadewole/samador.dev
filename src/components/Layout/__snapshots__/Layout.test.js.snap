// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render correctly 1`] = `
<div
  className="py-10 bg-gray-700 text-gray-300 dark:bg-gray-900 m-0 min-h-screen"
>
  <div
    className="xl:w-9/12 m-auto md:flex"
  >
    <div
      className="w-full lg:w-5/12 md:w-6/12 px-10 pb-10 border-b-2 md:border-b-0 md:border-r-2 border-gray-600"
    >
      <div
        className="mb-4"
      >
        <a
          href="/"
        >
          <img
            alt="Test name"
            className="rounded transform hover:scale-105"
            height="85"
            src="/test.jpg"
            width="85"
          />
        </a>
        <div />
        <h2
          className="text-2xl mb-2 leading-normal font-semibold text-white"
        >
          <a
            className=""
            data-testid="author"
            href="/"
          >
            Test name
          </a>
        </h2>
        <p
          className="text-sm font-light leading-relaxed mt-0 mb-4 text-gray-300"
        >
          Test bio
        </p>
      </div>
      <div
        className=""
      >
        <ul />
        <button
          aria-hidden="true"
          className={null}
          onClick={[Function]}
          style={
            Object {
              "MozAppearance": "none",
              "WebkitAppearance": "none",
              "appearance": "none",
              "backgroundColor": "transparent",
              "border": "none",
              "cursor": "pointer",
              "height": "37.599999999999994px",
              "overflow": "hidden",
              "padding": 0,
              "width": "80px",
            }
          }
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": "132px",
                "justifyContent": "center",
                "marginLeft": "-25.6px",
                "marginTop": "-47.599999999999994px",
                "width": "132px",
              }
            }
          >
            <div
              aria-label="animation"
              role="button"
              style={
                Object {
                  "height": "100%",
                  "margin": "0 auto",
                  "outline": "none",
                  "overflow": "hidden",
                  "width": "100%",
                }
              }
              tabIndex={0}
              title=""
            />
          </div>
        </button>
        <nav
          style={
            Object {
              "marginTop": "-50px",
            }
          }
        >
          <ul>
            <li
              className="text-gray-300"
            >
              -
               
              <a
                className="hover:underline font-light leading-snug mr-2"
                href="/test/1/"
              >
                Test label 1
              </a>
            </li>
            <li
              className="text-gray-300"
            >
              -
               
              <a
                className="hover:underline font-light leading-snug mr-2"
                href="/test/2/"
              >
                Test label 2
              </a>
            </li>
            <li
              className="text-gray-300"
            >
              -
               
              <a
                className="hover:underline font-light leading-snug mr-2"
                href="/test/3/"
              >
                Test label 3
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <main
      className="px-10 layout-body"
    >
      <h1
        className="md:text-5xl text-3xl md:mt-0 mt-8 mb-8"
      >
        test
      </h1>
      test
    </main>
  </div>
</div>
`;
